### PR TITLE
reusable ResultsTable component

### PR DESCRIPTION
**Pull request by kpsherva**

### CODE REVIEW

**Comment by zzacharo** [comment]
I think is really good! So when tests are passing I will review it again and then we can merge it!

File: invenio_app_ils/ui/src/invenio_app_ils/common/api/loan.js [hunk 1]
```diff
@@ -65,10 +67,24 @@ const fetchLoans = (
   return http.get(`${loanURL}?q=${qs}&sort:${sortByOrder}`);
 };
 
+const viewDetailsClickUrl = loanPid => {
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/common/api/loan.js]: I think the API layer shouldn't know anything about UI, like clicks or urls or views.
Why do you want to put this here? Probably better in `urls.js`, for example?

File: invenio_app_ils/ui/src/invenio_app_ils/common/api/loan.js [hunk 2]
```diff
@@ -65,10 +67,24 @@ const fetchLoans = (
...
+  const path = generatePath(BackOfficeURLS.loanDetails, {
+    loanPid: loanPid,
+  });
+  return path;
+};
+
+const showAllClickUrl = (documentPid, itemPid, state, patronPid) => {
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/common/api/loan.js]: Same for this I would say.

File: invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemPendingLoans/ItemPendingLoans.js [hunk 3]
```diff
@@ -1,180 +1,63 @@
...
-                position="top center"
-              />
-            </Table.HeaderCell>
-          </Table.Row>
-        </Table.Header>
-        <Table.Body>{_pendingLoans}</Table.Body>
-        {this._renderFooter(pendingLoans)}
-      </Table>
+    this.props.history.push(
+      this.showAllUrl(document_pid, item_pid, 'PENDING', null)
```
Comment (ntarocco) [inline invenio_app_ils/ui/src/invenio_app_ils/pages/backoffice/ItemDetails/components/ItemPendingLoans/ItemPendingLoans.js:34]: I would have generated the path here probably, it belongs to the UI layer.
If you need to re-use it, let's find a common place around URL handling.

**Comment by zzacharo** [comment]
@kprzerwa can you rebase and then we can merge it?

**Review by zzacharo — APPROVED** [review]

**Review by ntarocco — APPROVED** [review]

**Pull request merged**